src/coreclr/src/jit/codegenarm64.cpp

```diff
@@ -7864,6 +7858,40 @@ void CodeGen::genArm64EmitterUnitTests()
     theEmitter->emitIns_R_R(INS_fcvtn2, EA_8BYTE, REG_V0, REG_V1);
 #endif
 
+#ifdef ALL_ARM64_EMITTER_UNIT_TESTS
+    // sadalp vector
+    theEmitter->emitIns_R_R(INS_sadalp, EA_8BYTE, REG_V0, REG_V1, INS_OPTS_8B);
```

Comment:
These are a little odd because Vd and Vn take different arrangement specifiers, even though one implies the other. So which register does this insOpts specifier refer to? There must be other SIMD instructions like this as well. Should we have a function that takes two, and asserts they are as expected (e.g., if we have Vd.4H then we have Vn.8B)?

Are there instructions where the two specifies can both be arbitrary (and unrelated)?

Reply:
I would say most of the added "long" instructions (except high narrow - addhn{2}, subhn{2}) specify their source register(s) arrangement specifiers. 

I don't think we want to specify two arrangements when emitting at the moment - this would complicate the logic in the intrinsic backend - you would not only need to infer the insOpts from an incoming argument/return value but also to compute the second arrangement somehow.

> Are there instructions where the two specifies can both be arbitrary (and unrelated)?
I have not seen such instructions - I hope there won't be any